contracts/TreasuryInsurance.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import '@openzeppelin/contracts/utils/Address.sol';
+import '@openzeppelin/contracts/utils/Context.sol';
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/token/ERC721/ERC721.sol';
+import './interfaces/IERC20Allowance.sol';
+import './Treasury.sol';
+
+// Treasury contract with insurance.
+contract TreasuryInsurance is Treasury, ERC721 {
+    mapping(uint256 => uint256) public compLimits;
```

Comment:
question: is `comp` in `compLimits` short form for compensation?

If the term is short hand, then I'll suggest expanding to the full term (as all the other variables use the full terminology)